src/app/view/components/dot-navigation/dot-navigation.service.ts

```diff
@@ -85,14 +92,20 @@ export class DotNavigationService {
         this.dotMenuService.reloadMenu().subscribe((menu: DotMenu[]) => {
             this.dotMenuService
                 .isPortletInMenu(
-                    this.dotRouterService.currentPortlet.id ||
-                    this.dotRouterService.getPortletId(this.location.hash)
+                    this.dotRouterService.currentPortlet.id || this.dotRouterService.getPortletId(this.location.hash)
                 )
                 .subscribe((isPortletInMenu: boolean) => {
                     if (!isPortletInMenu) {
-                        this.goToFirstPortlet().then(res => {
-                            this.setMenu(menu);
-                        });
+                        if (this.dotRouterService.previousSavedURL) {
+                            this.dotRouterService
+                                .gotoPortlet(this.dotRouterService.previousSavedURL, true)
+                                .then(res => {
+                                    // Menu update happens on NavigationEnd event
+                                    this.dotRouterService.previousSavedURL = null;
+                                });
+                        } else {
+                            this.goToFirstPortlet(); // Menu update happens on NavigationEnd event
+                        }
                     } else {
                         this.setMenu(menu);
```

Comment:
But also menu it's updating here...

Reply:
The subscription to the NavigationEnd, is needed, because we need to update the menu item once the route is served and when non public page. In the previous scenario we were asking for a menu update before the route was updated, and that was causing issues with the selected option. 

Regarding the menu update you mention in this comment, is only executed when we use the functionality of "Login as" or we add or remove a new item in the menu. So is not duplicate functionality, each serve in different scenarios.